cca.go

```diff
@@ -0,0 +1,122 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package stat
+
+import (
+	"errors"
+	"math"
+
+	"github.com/gonum/floats"
+	"github.com/gonum/matrix"
+	"github.com/gonum/matrix/mat64"
+)
+
+// CanonicalCorrelations returns the `canonical' correlations between
+// the corresponding pairs of canonical variables -- described by the
+// left and right eigenvectors of the canonical correlation matrix,
+// pvecs and qvecs respectively. The canonical correlation matrix is
+// calculated using the singular value decompositions of the centered
+// inputs, and these singular value decompositions, xsvd and ysvd, are
+// also returned.
+//
+// The slice weights is used to weight the observations. If weights is nil,
+// each weight is considered to have a value of one, otherwise the length of
+// weights must match the number of observations or CanonicalCorrelations will
+// panic.
+//
+// Canonical Correlations will panic if the inputs x and y do not have the same
+// number of rows, and their columns should be interepretable as two different
+// sets of measurements (variables) on the same observations (each row representing
+// an observation).
+//
+// The sphered data are Xc * Sx^(-1/2) and Yc * Sy^(-1/2) respectively, and the
+// correlation matrix between the sphered data is the canonical correlation matrix,
+// Sx^(-1/2) * Sxy * Sy^(-1/2). In cases where S^(-1/2) is ambiguous, S^(-1/2) is
+// taken to be E * D^(-1/2) * E' where S can be eigendecomposed as S = E * D * E'.
+// Xc and Yc denote the centered input data matrices x and y (column means subtracted
+// from each column), Sx and Sy denote the sample covariance matrices within x and y
+// respectively, and Sxy denotes the covariance matrix between x and y.
```

Comment:
Is this what you mean? Funny, I had it the other way around earlier, with the notation first. I could always change it back I don't really mind either way that much.